client/my-sites/current-site/index.jsx

```diff
@@ -159,9 +156,13 @@ const CurrentSite = React.createClass( {
 // TODO: make this pure when sites can be retrieved from the Redux state
 module.exports = connect(
 	( state, ownProps ) => {
+		const selectedSiteId = getSelectedSiteId( state );
 		const selectedSite = getSelectedSite( state );
+
 		return {
-			selectedSite: selectedSite || ownProps.sites.getPrimary()
+			selectedSiteId,
```

Comment:
My only worry here is that in most cases all mapped props will come from redux state (selectedSiteId will === selectedSite.ID) but in the odd case that `selectedSite: ownProps.sites.getPrimary()` selectedSiteId will not necessarily correlate with `selectedSite.ID` (read `sites.getPrimary().ID`).
I'm in favour of this PR since it fixes the issue and is another step in the right direction but it leaves me wondering if `getPrimary()` has much value at all? and could there be new regressions based on the potential miss-match of IDs?